taichi/ir/ir.h

```diff
@@ -423,14 +423,21 @@ class StmtField {
 template <typename T>
 class StmtFieldNumeric final : public StmtField {
  private:
+  T *value_ptr;
   T value;
 
  public:
-  explicit StmtFieldNumeric(T value) : value(value) {
+  explicit StmtFieldNumeric(T *value_ptr) : value_ptr(value_ptr), value() {
+  }
+
+  explicit StmtFieldNumeric(T value) : value_ptr(nullptr), value(value) {
   }
 
   bool equal(const StmtField *other_generic) const override {
     if (auto other = dynamic_cast<const StmtFieldNumeric *>(other_generic)) {
+      if (other->value_ptr && value_ptr) {
+        return *(other->value_ptr) == *value_ptr;
+      }
```

Comment:
What if only one of them is a pointer?
```suggestion
      } else if (other->value_ptr || value_ptr) {
        return false;
      }
```